### PR TITLE
feat(torch): torch margins and vine distributions

### PR DESCRIPTION
**Pull request by tvatter**
Third of four. The torch half of the vine-distribution stack: `TorchMargin` and `TorchVinedist`.

## What it buys

Every part is a `torch.nn.Module`, so one `.to(device)` moves the whole joint distribution, `state_dict` captures all of it, and

```python
loss = -dist.logpdf(y).mean()
loss.backward()          # reaches the marginal parameters
```

works end to end. `log_prob` is available as the torch-idiomatic spelling of `logpdf`.

## Why `TorchMargin` rebuilds its distribution

A `torch.distributions.Distribution` has no `.to()` and, held as a plain attribute, contributes nothing to a `state_dict`. So `TorchMargin` registers the *parameters* and rebuilds the distribution on every call — the shape `TorchBicop` already uses for its interpolation grid. `TorchMargin.from_distribution` lifts an existing distribution or a factory closing over its own tensors.

## Honest limits, refused at construction

Continuous families only, and only those implementing `cdf`. `Gamma` / `Chi2` are rescued by inverting `cdf` by bisection over the support; `Beta` / `StudentT` have no `cdf` to invert, and every discrete family is out. All of these raise when the margin is built, not at first evaluation.

Because torch has no derivative for `igamma`'s first argument, a `Gamma` margin is fittable by its own marginal likelihood but **not** end to end through a copula — one more argument for keeping the two-step estimator the default.

`TorchVinedist` insists on torch parts throughout: a SciPy margin or a compiled `Vinecop` would silently detach every gradient, so it refuses them rather than producing a model whose `backward()` reaches nothing.

## Parity

`TorchVinedist.logpdf` matches the NumPy `Vinedist` on the same fitted model to 1e-10 — the tolerance the torch-vs-C++ copula cascade already pins.

## Verification

`make check` and `make docs` locally; the torch test files run in CI, since this workstation's GPU also drives the maintainer's session.

### CODE REVIEW

**Comment by tvatter** [comment]
Superseded by #302: its commits are the first two in that PR, alongside TorchKde1d which is what finally makes TorchVinedist.from_data work. Closing in favor of that PR; nothing here is dropped, and its changelog bullets now cite #302.
